modules/caddyhttp/reverseproxy/healthchecks.go

```diff
@@ -342,7 +351,11 @@ func (h *Handler) doActiveHealthCheck(dialInfo DialInfo, hostAddr string, networ
 	if err != nil {
 		host = hostAddr
 	}
-	if h.HealthChecks.Active.Port != 0 {
+
+	if h.HealthChecks.Active.Upstream != "" {
+		u.Host = h.HealthChecks.Active.Upstream
+	} else if h.HealthChecks.Active.Port != 0 {
+		// ignore active health check port if active upstream is provided
```

Comment:
This comment is positioned confusingly, it's inside of the "yes we'll use the port" case but it's saying "we're ignoring the port". This should go before the condition block to explain the logic

Reply:
Can we clarify this behavior in the godoc comment somewhere? If upstream is specified, port field is ignored.

Reply:
@mholt thanks for the comment - can you direct me to where you would like the comment? Do you mean here? https://github.com/caddyserver/caddy/pull/6451/files#diff-23b6ca3ed370c5279841c40f9687ff0ce19cea5e6ca4f3a688bbcbcde8c33a37L78-L80

Reply:
Sorry -- yes! That spot, and the new `Upstream` field above it might be good spots.

Reply:
Thanks, will do so now.